### PR TITLE
Bad regex in CORS settings should throw a nicer error

### PR DESCRIPTION
**Pull request by romseygeek**
Currently a bad regex in CORS settings throws a PatternSyntaxException, which
then bubbles up through the bootstrap code, meaning users have to parse a
stack trace to work out where the problem is. We should instead catch this
exception and rethrow with a more useful error message.

### CODE REVIEW

File: modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java [hunk 1]
```diff
@@ -241,11 +243,16 @@ static Netty4CorsConfig buildCorsConfig(Settings settings) {
...
-                builder = Netty4CorsConfigBuilder.forPattern(p);
+            try {
+                Pattern p = RestUtils.checkCorsSettingForRegex(origin);
+                if (p == null) {
+                    builder = Netty4CorsConfigBuilder.forOrigins(RestUtils.corsSettingAsArray(origin));
+                } else {
+                    builder = Netty4CorsConfigBuilder.forPattern(p);
+                }
+            }
+            catch (PatternSyntaxException e) {
```
Comment (danielmitterdorfer) [inline modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java]: Nit: `catch` should be on the same line as the closing brace of the `try` block.

File: modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java [hunk 2]
```diff
@@ -241,11 +243,16 @@ static Netty4CorsConfig buildCorsConfig(Settings settings) {
...
+                throw new SettingsException("Bad regex in " + SETTING_CORS_ALLOW_ORIGIN.getKey() + ": " + origin, e);
```
Comment (danielmitterdorfer) [inline modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java]: Can you please enclose the parameters in square brackets? i.e.:

```java
throw new SettingsException("Bad regex in [" + SETTING_CORS_ALLOW_ORIGIN.getKey() + "]: [" + origin + "]", e);
```

File: plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java [hunk 3]
```diff
@@ -176,11 +178,16 @@ static NioCorsConfig buildCorsConfig(Settings settings) {
...
-                builder = NioCorsConfigBuilder.forPattern(p);
+            try {
+                Pattern p = RestUtils.checkCorsSettingForRegex(origin);
+                if (p == null) {
+                    builder = NioCorsConfigBuilder.forOrigins(RestUtils.corsSettingAsArray(origin));
+                } else {
+                    builder = NioCorsConfigBuilder.forPattern(p);
+                }
+            }
+            catch (PatternSyntaxException e) {
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java]: Nit: `catch` should be on the same line as the closing brace of the `try` block.

File: plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java [hunk 4]
```diff
@@ -176,11 +178,16 @@ static NioCorsConfig buildCorsConfig(Settings settings) {
...
+                throw new SettingsException("Bad regex in " + SETTING_CORS_ALLOW_ORIGIN.getKey() + ": " + origin, e);
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java]: Can you please enclose the parameters in square brackets?

**Review by danielmitterdorfer — COMMENTED** [review]
Looks fine but I left some suggestions.

**Review by danielmitterdorfer — APPROVED** [review]
Thanks @romseygeek! LGTM

**Comment by romseygeek** [comment]
retest this please

**Pull request merged**